core/primitives/src/crypto/aggregate_signature.rs

```diff
@@ -276,19 +275,19 @@ impl<E: Engine> ToBytes for Signature<E> {
 }
 
 impl<E: Engine> TryFrom<&[u8]> for Signature<E> {
-    type Error = Box<Error>;
```

Comment:
This PR replaces many `type Error = Box<std::error::Error>;` with `type Error = String;`. What is the reason? We should using `type Error = Box<std::error::Error>;` wherever it is possible, and moving our code in that direction, not the other way around.

Reply:
What's the benefit of Box<std::error::Error>? I get custom error class with proper set of Froms to escalate errors, but why Box?

Reply:
Compatibility with third-party crates and std that almost always use structs that implement `std::error::Error`, e.g. io errors, network errors, etc. Boxing error allows us to pipe errors around the code without doing conversions. Additionally we can use `?` to make our code smaller. Without boxed error `?` will not work if some results inside the method are of different error type. Boxed error is a widespread pattern in Rust, enabled by `impl<T: Error> Error for Box<T>` in std https://doc.rust-lang.org/std/error/trait.Error.html . More info: https://doc.rust-lang.org/rust-by-example/error/multiple_error_types/boxing_errors.html

Reply:
Got it. Will do from now on.